### PR TITLE
[batch,ci,pipeline] change client - parent_ids -> parents

### PR DESCRIPTION
**Pull request by jigold**
I think this interface is clearer and it will be a necessary change when atomic batch creation causes the batch_id to be None until the entire batch is submitted.

### CODE REVIEW

**Comment by jigold** [comment]
Stacked on #6238

**Comment by akotlar** [comment]
Noted, will review when 6238 goes in.

File: pipeline/pipeline/backend.py [hunk 1]
```diff
@@ -283,16 +283,16 @@ def _cp(src, dst):
             return
 
         if delete_scratch_on_exit and used_remote_tmpdir:
-            parent_ids = list(job_id_to_command.keys())
+            parents = list(jobs_to_command.keys())
```
Comment (akotlar) [inline pipeline/pipeline/backend.py]: I think this is a really cool interface decision; it seems we don't need custom __hash__ and __eq__... but want to confirm (Python docs state that default __hash__ is id(), which is a stable identifier that is unique for the lifetime of the object).

Comment (jigold) [reply pipeline/pipeline/backend.py]: I'm pretty sure this is fine. https://stackoverflow.com/questions/4950155/objects-as-keys-in-python-dictionaries?rq=1

**Pull request merged**